### ISSUE TITLE
[Bug] Unable to Download Files

### ISSUE THREAD

**Issue by Lycaon37** [zurdi15/romm#268]
**RomM version**
1.84

**Describe the bug**
Downloaded files return as 524 bytes, regardless of the actual size.

Contents of a downloaded file are here, viewed via notepad. All downloaded files appear to have similar contents to this one.

```
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8" />
  <link rel="icon" href="/favicon.ico"/>
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>RomM</title>
  <script type="module" crossorigin src="/assets/index.83f1f892.js"></script>
  <link rel="stylesheet" href="/assets/index.e8039371.css">
<link rel="manifest" href="/manifest.webmanifest"><script id="vite-plugin-pwa:register-sw" src="/registerSW.js"></script></head>

<body>
  <div id="app"></div>
  
</body>

</html>
```

**To Reproduce**
1. Go to the RomM page.
2. Click on any game platform.
3. Click on a download button for a game.
4. Returned file is not the file expected.

**Expected behavior**
I expect the returned file to be the ROM that I have stored in the folder mounted to /romm/library

**Desktop (please complete the following information):**
 - Host OS: Synology DSM 7 (DS 920+ NAS) via Docker v20.10.3-1308
 - Client OS: Windows 11 Home 22H2
 - Browser: Mozilla Firefox
 - Version: 113.02

**Additional context**
I've tried accessing the site via both the mapped port and a reverse proxy (established through the Synology DSM reverse proxy system). The method that gave this error was accessing directly via the mapped port.

**Comment by GAntoine** [comment]
Could you post the request and response body from the chrome/ff developer tools?

**Comment by Lycaon37** [comment]
Headers:
![Screenshot 2023-05-27 134048](https://github.com/zurdi15/romm/assets/7057623/f6558efd-f63c-4281-a3d7-e21eb246196d)

Response Body:
![Screenshot 2023-05-27 134130](https://github.com/zurdi15/romm/assets/7057623/df048b6a-8f0e-4233-aa74-de7498c07715)

**Comment by GAntoine** [comment]
I see `Server: nginx` in the response headers, are you using nginx as your reverse proxy? And if so, could you post the config section relevant to romm?

I can also see the content-type is wrong, you'd expect to see something like `application/zip` for zip files.

**Comment by Lycaon37** [comment]
I'm using the Synology Reverse Proxy system, which uses nginx internally. Although for romm here I'm just connecting directly to it at the domain name with the mapped port. I tried connecting directly via the NAS's IP address and the mapped port, but I got the same issue. I'm not sure what Synology uses for nginx in terms of configuration.

**Comment by GAntoine** [comment]
Hmmm that's as far as my knowledge goes. The request is right, but your server is responding incorrectly. If romm is the cause, @zurdi15 might have an idea.

**Comment by calebsmithdev** [comment]
Confirming that I am having this issue as well with any file size. It looks like my HTML is the same as well. I am failing to download any size file, regardless of direct IP access or behind my reverse proxy.

**Comment by GAntoine** [comment]
>  I am failing to download any size file, regardless of direct IP access or behind my reverse proxy.

@calebsmithdev is your setup the same or similar, like on a Synology NAS?

**Comment by calebsmithdev** [comment]
I run mine through a Docker Compose - very similar to the example in the repo. My assumption is that the library path either isn't loading right, or it's not bypassing the routing on the frontend like it does for the resources path. I'll look at it more tomorrow to see about the library path. It was at least loading fine a couple of weeks ago.

**Comment by calebsmithdev** [comment]
Hmm. Well trying to rescan my library is giving me an error - whether I select a platform or not. I haven't added any new ROMs since the last update, so I may need to check something on my end.

`"GET /scan?platforms=[]&complete_rescan=false HTTP/1.0" 404 Not Found`
`"GET /scan?platforms=[%22gba%22]&complete_rescan=false HTTP/1.0" 404 Not Found`

**Comment by Lycaon37** [comment]
I tried running romm on my Windows 11 machine using the same docker compose file outside of changing the volume paths on the host end. That seemed to work normally, giving me the actual file.

Here's the headers for both:
**Windows**
![Screenshot (6)](https://github.com/zurdi15/romm/assets/7057623/cb0ca0af-a7a0-4bd4-9572-7241a53223b3)
**Synology**
![Screenshot (7)](https://github.com/zurdi15/romm/assets/7057623/291d9c26-9115-4019-a6f9-3270092fb5e1)

**Comment by GAntoine** [comment]
Well now that's interesting...

Can you try to `docker exec` into both running containers and see if the library is mounted correctly?

`docker exec -it romm bash`

And could you also post the relevant section from your docker-compose?

**Comment by Lycaon37** [comment]
**docker-compose.yml contents:**
```
version: '3'
volumes:
  mysql_data:
services:
  romm:
    image: zurdi15/romm:latest
    container_name: romm
    environment:
      - ROMM_DB_DRIVER=sqlite # This variable can be set as: mariadb | sqlite. If it is not defined, sqlite will be the database by default
      - CLIENT_ID=*IGDB CLIENT ID*
      - CLIENT_SECRET=*IGDB CLIENT SECRET*
      - STEAMGRIDDB_API_KEY=*STEAMGRIDDB API KEY*
    volumes:
      - '/volume1/docker/romm/library:/romm/library/roms'
    ports:
      - 3080:80
    restart: "unless-stopped"
```

Windows docker exec:
![Screenshot 2023-05-29 094144](https://github.com/zurdi15/romm/assets/7057623/f12da80d-e26d-44f1-8d8d-2f1be211496e)

Synology NAS docker exec
![Screenshot 2023-05-29 094450](https://github.com/zurdi15/romm/assets/7057623/fbe203bc-1c00-4510-a5f6-0361b5d20bae)

**Comment by GAntoine** [comment]
I have a couple working theories based on everything I've seen so far.

a) Something funky is happening with certs(?): when fetching from localhost, the request has 🔒 , but when fetching from the ip address, the lock has a strike through it 🔏
b) Browser cache: try clearing cache and cookies for the Synology version
c) Last resort: Stop and remove the container, purge any volumes and start it up again. **THIS WILL CLEAR YOUR LIBRARY AND YOU'LL NEED TO RESCAN**

**Comment by Lycaon37** [comment]
It seems the crossed out padlock appeared because Firefox was connecting over http to an IP address instead of over HTTPS. It's not crossed out for localhost because it's connecting to the same computer instead of a network. When I connect via my reverse proxy, the issue remains, but the lock appears normal as it's over HTTPS.

I tried clearing the cache and cookies on Firefox and Edge, and both had the same issue. Removing the container, purging the volumes, and then starting the container again with docker-compose didn't seem to affect the issue.

I used these commands to clear the container:
```
docker-compose down
docker volume prune
docker-compose up -d 
```
